core/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/Decision.java

```diff
@@ -82,9 +82,9 @@ public static Decision readFrom(StreamInput in) throws IOException {
      * possible types of decisions
      */
     public enum Type implements Writeable {
-        NO(0),
         YES(1),
-        THROTTLE(2);
+        THROTTLE(2),
```

Comment:
now you rely on the natural order? You should then also have a test for it so that it does not get reordered by anyone else.